java/src/org/openqa/selenium/grid/web/PrefixAwareJarFileResource.java

```diff
@@ -0,0 +1,80 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
```

Comment:
This class duplicates stuff. Instead can we please consider refactoring this class into something like this ?

```java
package org.openqa.selenium.grid.web;

import java.util.Arrays;
import java.util.List;
import java.util.Optional;
import java.util.jar.JarFile;

public class PrefixAwareJarFileResource extends JarFileResource {
    private final String subPath;

    public PrefixAwareJarFileResource(JarFile jarFile, String entryName, String stripPrefix, String subPath) {
        super(jarFile, entryName, stripPrefix);
        this.subPath = subPath;
    }

  @Override
  protected JarFileResource newInstance(JarFile jarFile, String entryName, String prefix) {
    return new PrefixAwareJarFileResource(jarFile, entryName, prefix, subPath);
  }

  private static final List<String> EXCLUSION_LIST = Arrays.asList(
    ".woff", ".woff2", ".svg", ".png"
  );

    @Override
    public Optional<byte[]> read() {
      Optional<byte[]> readByteArray = super.read();
      boolean found = EXCLUSION_LIST.parallelStream()
        .anyMatch(entryName::endsWith);
      if ( found) {
        System.err.println(">>>> Skipping entry [" + entryName + "] <<<<");
        return readByteArray;
      }
      System.err.println("Processing entry :::: [" + entryName + "] ::::");
      if (!readByteArray.isPresent()) {
        return readByteArray;
      }
      String readString = new String(readByteArray.get());

      // All file such as index.html, *.js, *.css and *.js.map
      readString = readString.replaceAll("/ui/", subPath.concat("/ui/"))
        .replaceAll("\\+\"/graphql\"", "+\"".concat(subPath).concat("/graphql\""));

      return Optional.of(readString.getBytes());
    }
}
```

Reply:
Done the changes.